src/Shared/Debugging/DebugUtils.cs

```diff
@@ -22,12 +23,27 @@ private enum NodeMode
 
         static DebugUtils()
         {
-            string environmentDebugPath = Environment.GetEnvironmentVariable("MSBUILDDEBUGPATH");
-            var debugDirectory = environmentDebugPath;
+            string environmentDebugPath = FileUtilities.TrimAndStripAnyQuotes(Environment.GetEnvironmentVariable("MSBUILDDEBUGPATH"));
+            string debugDirectory = environmentDebugPath;
+
+            DebugPathMessageImportance = MessageImportance.Low;
 
             if (Traits.Instance.DebugEngine)
             {
-                debugDirectory ??= Path.Combine(Directory.GetCurrentDirectory(), "MSBuild_Logs");
+                DebugPathMessageImportance = MessageImportance.High;
+                if (!string.IsNullOrWhiteSpace(debugDirectory) && FileUtilities.CanWriteToDirectory(debugDirectory))
+                {
+                    // Debug directory is writable; no need for fallbacks
+                    DebugPathMessageImportance = MessageImportance.Low;
```

Comment:
I think I'd prefer to leave this as high: it's interesting that they've used the env var to set up logging even if it works.